src/Modules/homological-algebra.jl

```diff
@@ -1,4 +1,498 @@
+ if res == 1 return true end
+ return false
+end
+
+@doc Markdown.doc"""
+     non_flat_locus(M::ModuleFP) 
+
+Return an ideal of `base_ring(M)` which defines the non-flat-locus of `M`
+in the sense that the localization of `M` at a prime ideal of `base_ring(M)`
+is non-flat iff the prime ideal contains the returned ideal.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = PolynomialRing(QQ, ["x", "y"]);
+
+julia> F = free_module(R, 2);
+
+julia> o = zero(R)
+0
+
+julia> U = matrix([x^3-y^2 o; o x^3-y^2; -x^2 y; -y x])
+[x^3 - y^2           0]
+[        0   x^3 - y^2]
+[     -x^2           y]
+[       -y           x]
+
+julia> M = quo(F,U)[1]
+Subquotient of Submodule with 2 generators
+1 -> e[1]
+2 -> e[2]
+by Submodule with 4 generators
+1 -> (x^3 - y^2)*e[1]
+2 -> (x^3 - y^2)*e[2]
+3 -> -x^2*e[1] + y*e[2]
+4 -> -y*e[1] + x*e[2]
+
+julia> non_flat_locus(M)
+ideal(x^3 - y^2)
+```
+"""
+function non_flat_locus(M::ModuleFP)
+ error("not implemented for the given type of module.")
+end
+
+function non_flat_locus(F:: FreeMod)
+ R = base_ring(F)
+ if !(coefficient_ring(R) isa AbstractAlgebra.Field)
+   throw(ArgumentError("The coefficient ring must be a field."))
+ end
+ return ideal(R, [one(R)])
+end
+
+function non_flat_locus(M::SubQuo)
+ R = base_ring(M)
+ if !(coefficient_ring(R) isa AbstractAlgebra.Field)
+   throw(ArgumentError("The coefficient ring must be a field."))
+ end
+ C = present_as_cokernel(M)
+ U = C.quo
+ SG = singular_generators(U.gens)
+ return MPolyIdeal(R, Singular.LibHomolog.flatLocus(SG))
+end
+
+##############################################################################
+#
+# Regular sequence test
+#
+##############################################################################
+
+@doc Markdown.doc"""
+     is_regular_sequence(V::Vector{T}, M::ModuleFP{T}) where T <: MPolyElem
+
+Return `true` if the elements of `V` form, in the given order, a regular sequence on `M`.
+Return `false`, otherwise.
+
+# Examples
+```jldoctest
+julia> R, (x, y, z) = PolynomialRing(QQ, ["x", "y", "z"]);
+
+julia> F = free_module(R, 1);
+
+julia> V =  [x*z-z, x*y-y, x]
+3-element Vector{fmpq_mpoly}:
+ x*z - z
+ x*y - y
+ x
+
+julia> is_regular_sequence(V, F)
+false
+
+julia> W = [x*z-z, x, x*y-y]
+3-element Vector{fmpq_mpoly}:
+ x*z - z
+ x
+ x*y - y
+
+julia> is_regular_sequence(W, F)
+true
+```
+"""
+function is_regular_sequence(V::Vector{T}, M::ModuleFP{T}) where T <: MPolyElem
+ error("not implemented for the given type of module.")
+end
+
+function is_regular_sequence(V::Vector{T}, F:: FreeMod{T}) where T <: MPolyElem
+ R = base_ring(F)
+ if !(coefficient_ring(R) isa AbstractAlgebra.Field)
+   throw(ArgumentError("The coefficient ring must be a field."))
+ end
+ @assert parent(V[1]) == R
+ @assert all(x->parent(x) == R, V)
+ I = ideal(R, V)
+ singular_assure(I)
+ MX = singular_module(F)
+ SG = Singular.Module(base_ring(MX), MX(zero(F)))
+ res = Singular.LibHomolog.isReg(I.gens.S, SG)
+ if res == 1 return true end
```

Comment:
```suggestion
one(R) in res && return true
```
Just another short way to write the same thing; not necessarily better but closer to what was once suggested to me to avoid the if-clauses.